recipes-support/dnsmasq/patches/dnsmasq_dnsout_interface.patch

```diff
@@ -102,3 +75,31 @@
      case 'I':  /* --except-interface */
      case '2':  /* --no-dhcp-interface */
        do {
+--- a/src/forward.c
++++ b/src/forward.c
+@@ -2162,9 +2162,25 @@ static struct frec *allocate_frec(time_t
+ static int random_sock(struct server *s)
+ {
+   int fd;
++  struct ifreq ifr;
++  struct iname *ifn;
++
++  memset(&ifr, 0, sizeof (ifr));
++  if ((ifn = daemon->dnsout_names))
++      strncpy(ifr.ifr_name, ifn->name, sizeof (ifr.ifr_name));
+ 
+   if ((fd = socket(s->source_addr.sa.sa_family, SOCK_DGRAM, 0)) != -1)
+     {
++            if (ifr.ifr_name[0])
++            {
++                printf("Use interface %s\n", ifr.ifr_name);
++                if (setsockopt(fd, SOL_SOCKET, SO_BINDTODEVICE,
++                               (void *)&ifr, sizeof (ifr)) < 0) 
++			    {
```

Comment:
Generally looks good, but you let some tabs slip in here while the rest of the file is spaces.

Reply:
The source file (forward.c) seems to be mixed tabs and spaced.  Usually 1-2 tabs depending on indent level, then spaces to align.  it's pretty awful.

Reply:
I could update the patch to only use spaces but technically it wouldn't follow what forward.c is doing elsewhere

Reply:
It also looks like the meta-oe backport https://github.com/openembedded/meta-openembedded/blob/2e7e98cd0cb82db214b13224c71134b9335a719b/meta-networking/recipes-support/dnsmasq/dnsmasq/CVE-2021-3448.patch made a mess of spaces and tabs.

Still, I think our addition should at least be internally consistent.  So do your best?

Reply:
ack, one sec